dataikuapi/fm/instances.py

```diff
@@ -37,14 +37,12 @@ def with_dss_node_type(self, dss_node_type):
         """
         Set the DSS Node type of the instance to create
 
-        :param str dss_node_type: Optional , the type of the dss node to create. Supports "design", "automation ordeployer". Defaults to "design"
+        :param dss_node_type: Optional , the type of the dss node to create. Defaults to `DESIGN`.
```

Comment:
I don't know if we want to be compatible with old usage, but with this change it will break (AttributeError: 'str' object has no attribute 'value')

If we do not want to break, either:
- cover the two types with a check: isinstance(dss_node_type, FMNodeType) 
- just add a string govern instead
- have the string value builder outside of this method

Quick note: maybe `def with_dss_node_type(self, dss_node_type=FMNodeType.DESIGN):` in the signature to make it "really" optional as a technical method or dss_node_type="design" for string

Reply:
> Quick note: maybe `def with_dss_node_type(self, dss_node_type=FMNodeType.DESIGN):` in the signature to make it "really" optional as a technical method or dss_node_type="design" for string

the optional is not welcome in the doc , as the parameter is not optional .. it's the method call that is optional, as it has a default value of DESIGN if you do not use `with_dss_node_type` . there is another story about cleaning comments though ..

Reply:
> I don't know if we want to be compatible with old usage

not sure it is needed, as I doubt customers knows there was an API before .. maybe it's better to have clean script now . my 2 cents .. Fred will tell us its thoughts hopefully :-)

Reply:
Ah yes, indeed, in this case, I would put the defaults to design only in the other method that actually put the default to design and I would remove the word optional from here in the doc. Other it may be misleading for this particular method usage.

Noted, and ok pour waiting for the compatibility